app/src/org/commcare/activities/ReminderThread.java

```diff
@@ -0,0 +1,139 @@
+                }catch (Exception e){
+                    e.printStackTrace();
+                    return;
+                }
+
+                //Prepare evaluation context
+                Hashtable<String, DataInstance> instances = new Hashtable<>();
+                instances.put("casedb", new ExternalDataInstance("jr://instance/casedb","casedb"));
+                EvaluationContext ec = CommCareApplication._().getCurrentSessionWrapper().getEvaluationContext(instances);
+
+                //Prepare TreeReferences
+                String caseXPathRef = "instance('casedb')/casedb/case[@case_type='" + caseTypeString + "']";
+                TreeReference caseType = XPathReference.getPathExpr(caseXPathRef).getReference(); //This is more of an abstract reference to a bunch of stuff in the tree
+                List<TreeReference> refs = ec.expandReference(caseType); //When we expand the reference within the context ec we get the x number of references that actually "exist"
+
+                while(mContinuePolling) {
+                    long current = System.currentTimeMillis();
+
+                    //Evaluate condition
+                    for(TreeReference ref : refs) {
+                        EvaluationContext internal = new EvaluationContext(ec, ref);
+                        Log.d("Result", condition.eval(internal).toString());
+
+                        if((Boolean) condition.eval(internal)){
+                            playNoise();
+                        }
+                    }
+                    
+                    long delay = (System.currentTimeMillis() - current);
+                    
+                    try {
+                        Thread.sleep(sleepTime * 1000 - delay);
+                        System.out.println("Sleeping for" + String.valueOf(sleepTime*1000-delay));
+                        sleepTime *= 2;
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                        mContinuePolling = false;
+                    }
+                }
+            }
+            
+            
+        };
+        
+        myThread = new Thread(runnable);
+    }
+    
+    public void playNoise() {
+        try {
+            Uri notification = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
+            Ringtone r = RingtoneManager.getRingtone(mContext, notification);
+            r.play();
+        } catch(Exception e) {
+            e.printStackTrace();
+        }
+    }
+    
+    public void startPolling(Vector<Alert> alerts) {
+
+        for(Alert a: alerts){
+            caseTypeString = a.getCaseType();
+            xPathConditionString = a.getCondition();
+        }
+
+        synchronized(myThread) {
+            if(!myThread.isAlive()) {
+                myThread.start();
+            }
+        }
+    }
+    
+    public void stopService() {
+        synchronized(myThread) {
+            mContinuePolling = false;
```

Comment:
It would be helpful to interrupt the thread inside of this as well to force it to process the threat stoppage.